File: Constructorio_Net/src/models/RedirectData.java
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

/**
 * Constructor.io Redirect Data
 */*/
namespace Constructorio_NET
{
    public class RedirectData {

        [JsonPropertyName("url")]
        public String url;

        [JsonPropertyName("rule_id")]
        public Integer ruleId;

        [JsonPropertyName("match_id")]
        public Integer matchId;

    }
}

File: src/constructor.io/models/Search/RedirectData.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

/**
 * Constructor.io Redirect Data
 */
namespace Constructorio_NET.Models
{
    public class RedirectData
    {
        [JsonProperty("url")]
        public string Url { get; set; }

        [JsonProperty("rule_id")]
        public int RuleId { get; set; }

        [JsonProperty("match_id")]
        public int MatchId { get; set; }

        [JsonExtensionData]
        private Dictionary<string, object> _additionalData;

        /// <summary>
        /// Indexer for accessing arbitrary metadata keys
        /// </summary>
        public object this[string key]
        {
            get => _additionalData?.TryGetValue(key, out var value) == true ? value : null;
            set
            {
                _additionalData ??= new Dictionary<string, object>();
                _additionalData[key] = value;
            }
        }
    }
}